src/sdk/cohort.ts

```diff
@@ -0,0 +1,77 @@
+import { PublicKey } from '@nucypher/nucypher-core';
+
+import { Porter } from '../characters/porter';
+import { ChecksumAddress } from '../types';
+
+export type Ursula = {
+  readonly checksumAddress: ChecksumAddress;
+  readonly uri: string;
+  readonly encryptingKey: PublicKey;
+};
+
+type CohortParameters = {
+  porterUri: string;
+  threshold: number;
+  shares?: number;
```

Comment:
yes, shares is optional, because a user can just supply a list of Ursula checksums instead - at which point shares is just the length of that array

Reply:
For global configs (i.e. accessible to all adopters), it makes more sense not to equate `shares` and `include_ursulas` – indeed, to move the latter to a different configuration stage so it's not being reused unknowingly/inappropriately. 

The former should be specified in all cases, because it's compatible with static lists and with more complex downstream sampling mechanisms at the time of use (adding new recipients, encrypting new data, adjusting some other parameter etc.) AND can safely be reused by other users or applications who have the same views on relative collusion-resistance. 

The latter is a bit more nuanced – the primary use of the `include_ursulas` parameter will be on a per-app basis – i.e. only relevant and appropriate for a single use case and user base. Some unrelated hackathon project prioritizing the pinging of nodes run by Masterfile's media partners would be a strange pattern to facilitate. 

See this feedback in the context of mixing-and-matching known nodes and permissionless sampling:
_Yes, overall prefer the decentralized version, as long as the option exists to hand-pick later_ https://github.com/nucypher/tdec/issues/13#issuecomment-1211954228

Reply:
For global configs (i.e. accessible to all adopters), it makes more sense not to equate `shares` and `include_ursulas` – indeed, to move the latter to a different configuration stage so it's not being reused unknowingly/inappropriately. 

The former should be specified in all cases, because it's compatible with static lists and with more complex downstream sampling mechanisms at the time of use (adding new recipients, encrypting new data, adjusting some other parameter etc.) AND can safely be reused by other users or applications who have the same views on relative collusion-resistance. 

The latter is a bit more nuanced – the primary use of the `include_ursulas` parameter will be on a per-app basis – i.e. only relevant and appropriate for a single use case and user base. Some unrelated hackathon project prioritizing the pinging of nodes run by Masterfile's media partners would be a strange pattern to facilitate. 

See this feedback in the context of mixing-and-matching known nodes and permissionless sampling:
_Yes, overall prefer the decentralized version, as long as the option exists to hand-pick later_ https://github.com/nucypher/tdec/issues/13#issuecomment-1211954228

Reply:
Agreed. At the moment it feels like there may need to be a `CohortConfiguration` which is just configuration values and a `Cohort` that gets generated from the `CohortConfiguration` at the appropriate time. Then that `Cohort` object would contain actual Ursulas to use.

The `CohortConfiguration` could be the defined "good", "better", "best" recommendations.

I agree that include/exclude is too nuanced for global configurations (all apps). That being said Masterfile could have a its own `CohortConfiguration` that provides include/exclude for their own ecosystem.

You could allow new `CohortConfiguration` object to based on other ones -  `CohortConfiguration.fromConfiguration(global_config, <values to add/modify>)` - for example `CohortConfiguration.fromConfiguration(<other config>,  include_ursulas=[0x..., 0x...])`.

I am pro on CohortConfiguration being immutable, and new objects needing to be created to change values.

Reply:
Yeah I've slowly been persuaded :D